### PR TITLE
draw function and sortig layers to show drawn polygon on top

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: src/components/Download.vue [hunk 1]
```diff
@@ -70,7 +70,7 @@
           </v-btn>
           <v-btn
             class="mb-1"
-            :disabled="downloading || bbox.coordinates || map.getZoom() < 9"
+            :disabled="downloading || !bbox.coordinates || map.getZoom() < 9"
```
Comment (SiggyF) [inline src/components/Download.vue]: Does `bbox` exist

File: src/components/Download.vue [hunk 2]
```diff
@@ -131,46 +131,65 @@ export default {
...
-      bbox: '',
+      bbox: {},
       draw: {},
       downloading: false,
       error: ''
     }
   },
   components: { SelectPeriod },
   mounted() {
+    console.log('mounted', JSON.parse(JSON.stringify(this.draw)))
```
Comment (SiggyF) [inline src/components/Download.vue]: please remove

File: src/components/Download.vue [hunk 3]
```diff
@@ -131,46 +131,65 @@ export default {
...
+      // add the control
+      this.map.addControl(this.draw, 'top-right')
+      // Now restore old loaded function
+      this.map.loaded = oldLoaded
+      this.map.on('draw.create', this.createFunc)
+      this.map.on('draw.update', this.updateFunc)
+      this.map.on('draw.delete', this.deleteFunc)
+    },
+    createFunc() {
+      console.log('create', this.draw)
```
Comment (SiggyF) [inline src/components/Download.vue]: please remove logging

File: src/components/Download.vue [hunk 4]
```diff
@@ -131,46 +131,65 @@ export default {
...
+    this.map.off('draw.update', this.updateFunc)
+    this.map.off('draw.delete', this.deleteFunc)
+    this.draw = {}
+  },
+  methods: {
+    createDrawTools() {
+      // loaded checks tiles. Video tiles are loading dynamicly
+      // temporary monkeypatch loaded function, because video layers are not always loaded
+      // keep the old loaded function
+      let oldLoaded = this.map.loaded
```
Comment (SiggyF) [inline src/components/Download.vue]: What is this?

File: src/components/Download.vue [hunk 5]
```diff
@@ -131,46 +131,65 @@ export default {
...
         this.draw.delete(oldId)
       }
+      console.log(this.draw)
       this.bbox = this.draw.getAll().features[0].geometry
-    })
-
-    this.map.on('draw.update', () => {
+      console.log(this.map.getStyle().layers)
+    },
+    updateFunc() {
```
Comment (SiggyF) [inline src/components/Download.vue]: func?

File: src/components/MapLayers.vue [hunk 6]
```diff
@@ -254,22 +254,28 @@ export default {
...
-        this.layerTypes.forEach(data => {
-          if (!this.filteredLayers[i][data]) return
-          for (
-            var thislayer = 0;
-            thislayer < this.filteredLayers[i][data].length;
-            ++thislayer
-          ) {
-            const mapId = this.filteredLayers[i][data][thislayer].id
+      this.filteredLayers.forEach((filterLayer, i) => {
+        this.layerTypes.forEach(type => {
```
Comment (SiggyF) [inline src/components/MapLayers.vue]: what are `filteredLayers` and  `layerTypes`

File: src/components/MapLayers.vue [hunk 7]
```diff
@@ -254,22 +254,28 @@ export default {
...
+              previousId = mapId
             }
-          }
+          })
         })
-      }
+      })
+    },
+    loadingMessage() {
+      return `De volgende lagen worden nog geladen: ${this.loadingLayers.join(', ')}.`
```
Comment (SiggyF) [inline src/components/MapLayers.vue]: I thought I removed this? Are you up to date?

File: src/components/Download.vue (hunk 1 above)
Comment (CindyvdVries) [reply src/components/Download.vue]: bbox is always an object

File: src/components/MapLayers.vue (hunk 6 above)
Comment (CindyvdVries) [reply src/components/MapLayers.vue]: filteredlayers are the layers that are used in this specific mode
layertypes are the types of layers expected in these layers (imagelayers, mapboxlayers etc)

File: src/components/MapLayers.vue (hunk 7 above)
Comment (CindyvdVries) [reply src/components/MapLayers.vue]: Oeps, my mistake. There was a conflict by merging and I chose the wrong one

**Pull request merged**